syncd/ComparisonLogic.cpp

```diff
@@ -1915,6 +1932,192 @@ void ComparisonLogic::processObjectForViewTransition(
+                {
+                    if (revgraph->attrmetadata->allownullobjectid)
+                    {
+                        // we can also remove entire object here too
+
+                        SWSS_LOG_THROW("break the link is not implemented yet, FIXME");
+                    }
+                    else
+                    {
+                        // attribute is create and set, but not allow null object id
+                        // so probably attribute is mandatory on create, we can't break
+                        // the link, but we can remove entire object
+
+                        removeCurrentObjectDependencyTree(currentView, temporaryView, obj); // recursion
+                    }
+                }
+                else if (revgraph->attrmetadata->iscreateandset && status == SAI_OBJECT_STATUS_MATCHED)
+                {
+                    SWSS_LOG_THROW("matched break the link is not implemented yet, FIXME");
+                }
+                else
+                {
+                    SWSS_LOG_THROW("remove on %s, obj status: %d, not supported, FIXME", revgraph->attrmetadata->attridname, status);
+                }
+            }
+        }
+    }
+
+    removeExistingObjectFromCurrentView(currentView, temporaryView, currentObj);
+}
+
+void ComparisonLogic::breakBeforeMake(
+        _In_ AsicView &currentView,
+        _In_ AsicView &temporaryView,
+        _In_ const std::shared_ptr<SaiObj>& currentBestMatch, // can be nullptr
+        _In_ const std::shared_ptr<SaiObj>& temporaryObj)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * Break Before Make rule.
+     *
+     * We can have 2 paths here:
+     *
+     * - current best match object was not found (nullptr), in case of limited
+     *   resources, this mean that we want to remove some existing object
+     *   before creating new, so number of existing objects will not exceed
+     *   initial value, for example we remove acl table, then we create acl
+     *   table.  Of course this will not hold true, if temporary number of
+     *   objects of given type is greater than current existing objects.
+     *
+     * - current best match was found, but perform Object Set Transition
+     *   operation failed, so we can't move current object attributes to
+     *   temporary object attributes, because for example some are create only.
+     */
+
+    if (!m_breakConfig->shouldBreakBeforeMake(temporaryObj->getObjectType()))
+    {
+        // skip object if not in break config
+
+        return;
+    }
+
+    if (currentBestMatch == nullptr)
+    {
+        // it can happen that current best match is not found for temporary
+        // object then we need to find best object for removal from existing
+        // ones the most suitable should be the one with most same attributes,
+        // since maybe only one read only attribute has been changed, and this
+        // will automatically result in null best match
```

Comment:
Let's say we have two objects with identical attributes (a good example in SONiC would be an L3 ACL table and a MIRROR ACL table), and then between versions we add one attribute to the MIRROR table. Are we guaranteed that this new attribute will be added to the MIRROR table and not the L3 table? How does that reconciliation happen?

Reply:
not sure if i understand your question, if you have 2 different objects, and then add another attribute to one of them, comparison logic make sure that this mirror object will have this new attribute, and object id will be set on other objects as compiled by orch agent. entire comparison logic process is complicated to describe it here, if you want to find out more we can have online call to discuss this

Reply:
I think I get your point, but I would still like to pick your brain about this sometime. :)

Reply:
sure :)